### PR TITLE
Phase2-hgx232 Further attempt to make dd4hep compliant

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Further attempt to make HGCalParameter to be populated by the xml with dd4hep interface

#### PR validation:

Tested  with cfg files in Geometry/HGCalCommonData/test/python

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h [hunk 1]
```diff
@@ -79,20 +77,20 @@ class HGCalGeomParameters {
...
-                           const std::vector<int32_t>& wafer2copy,
-                           const std::vector<HGCalGeomParameters::cellParameters>& wafers,
-                           const std::map<int, int>& wafertype,
-                           const std::map<int, HGCalGeomParameters::cellParameters>& cellsf,
-                           const std::map<int, HGCalGeomParameters::cellParameters>& cellsc,
-                           HGCalParameters& php);
-  void loadGeometryHexagon8(const std::map<int, HGCalGeomParameters::layerParameters>& layers,
-                            std::map<std::pair<int, int>, HGCalParameters::hgtrform>& trforms,
-                            const int& firstLayer,
+  void loadGeometryHexagon(const std::map<int, HGCalGeomParameters::layerParameters>&,
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: what is the purpose of this change?

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc [hunk 2]
```diff
@@ -384,13 +386,13 @@ void HGCalGeomParameters::loadGeometryHexagon(cms::DDFilteredView& fv,
             double zv[2], rv;
             const std::vector<double>& pars = fv1.parameters();
             if (mode == HGCalGeometryMode::Polyhedra) {
-              zv[0] = pars[4];
-              zv[1] = pars[7];
-              rv = pars[6];
+              zv[0] = pars[3];
```
Comment (kpedro88) [inline Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: These indices seem almost random. Some comments or more descriptive naming would be helpful to understand what is happening.

Comment (cvuosalo) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: @kpedro88 It's on my task list to create a shape class for Polyhedra (and some other shapes) to hide these implementation details. Unfortunately, I probably won't get to it soon. In the interim, we have these details here. Some comments saying that this code will be revised might help.

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: Header file need not know the name of the variable. It is the case for declaration of many methods. Otherwise this change does not add to anything

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: I talked to Ianna about this. May be we should discuss about this in some meeting.

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h (hunk 1 above)
Comment (makortel) [reply Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: It is nevertheless considered a good habit to name the parameters in the header for better self-documentation of the interface. Actually our code rule 4.27 also recommends that "Provide meaningful argument names in method declarations in the header ﬁle to indicate usage, unless the type fully describes the usage."

Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: Doing as suggested

File: Geometry/HGCalCommonData/src/HGCalGeomParameters.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalGeomParameters.cc]: @ianna Please comment on this issue

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by makortel** [comment]
> Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

Just to note that these workflows are broken in the IB (and fix has been merged already).

**Comment by smuzaffar** [comment]
please test
new IB with fix is now available

**Comment by kpedro88** [comment]
+upgrade

**Comment by ianna** [comment]
+1

**Comment by silviodonato** [comment]
please test workflow 23434.1001

**Comment by silviodonato** [comment]
+1

File: Geometry/HGCalCommonData/interface/HGCalGeomParameters.h [hunk 3]
```diff
@@ -30,39 +30,50 @@ class HGCalGeomParameters {
...
-                           const std::string&,
-                           const std::string&,
-                           const std::string&,
-                           const std::string&);
-  void loadSpecParsHexagon8(const DDFilteredView&, HGCalParameters&);
-  void loadSpecParsHexagon8(const cms::DDFilteredView&, const cms::DDVectorsMap&, HGCalParameters&, const std::string&);
-  void loadSpecParsTrapezoid(const DDFilteredView&, HGCalParameters&);
-  void loadSpecParsTrapezoid(const cms::DDFilteredView&, const cms::DDVectorsMap&, HGCalParameters&, const std::string&);
+
+  void loadGeometryHexagon(const DDFilteredView& _fv,
```
Comment (vargasa) [inline Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: Hello @bsunanda, I see you are loading `DDFilteredView/CompactView` both from `Core` and `DDCMS`. Do you have a list of missing methods (or compiler errors) that you get by using `DDCMS` instances only?

Comment (bsunanda) [reply Geometry/HGCalCommonData/interface/HGCalGeomParameters.h]: @vargasa The idea is that I would like to load geometry for either DDD or DD4Hep scenario. The difference is created by a run time flag which is under control of an ERA. So when this flag is set it loads geometry using DDCMS call, otherwise using call to Core. The 2 scenarios DDD vs DD4Hep need to coexist for a while. I do not want to have 2 libraries for core and DDCMS functionalities at this moment.
